dd4t-core/src/main/java/org/dd4t/core/factories/impl/PageFactoryImpl.java

```diff
@@ -127,9 +129,9 @@ public Page findPageByUrl (String url, int publicationId) throws FactoryExceptio
 		CacheElement<Page> cacheElement = cacheProvider.loadPayloadFromLocalCache(cacheKey);
 		Page page;
 
-		if (cacheElement.isExpired()) {
-			synchronized (cacheElement) {
-				if (cacheElement.isExpired()) {
+		if (cacheElement.isExpired() || cacheElement.getPayload() == null) {
+			synchronized (CACHE_LOCK) {

```

Comment:
This will ensure that each factory request is handled 1 at the time.. Not handy for a multithreaded web environment and should be restored to what it was..